homeassistant/components/sensor/foobot.py

```diff
@@ -0,0 +1,147 @@
+"""
+Support for the Foobot indoor air quality monitor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.foobot/
+"""
+import asyncio
+import logging
+from datetime import timedelta, datetime
+from collections import OrderedDict
+
+import aiohttp
+import voluptuous as vol
+
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    ATTR_TIME, ATTR_TEMPERATURE, CONF_TOKEN, CONF_USERNAME, TEMP_CELSIUS)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['foobot_async==0.3.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_HUMIDITY = 'humidity'
+ATTR_PM2_5 = 'PM2.5'
+ATTR_CARBON_DIOXIDE = 'CO2'
+ATTR_VOLATILE_ORGANIC_COMPOUNDS = 'VOC'
+ATTR_FOOBOT_INDEX = 'index'
+
+SENSOR_TYPES = OrderedDict([('time', [ATTR_TIME, 's']),
```

Comment:
Does this have to be an `OrderedDict`? As far as I see, a normal `dict` should also work.. 👀

Reply:
Having an `OrderedDict` made sense in the initial version I made as the API response is pretty much a JSON encapsulated CSV format and when walking through results, it was important that it was done in same order. 

Since then I abstracted that in the api library package and it didn't serve any purpose, I changed it back to a plain `Dict`.